framework/sandstone.cpp

```diff
@@ -660,6 +660,37 @@ inline void test_the_test_data<true>::test_tests_finish(const struct test *the_t
 #undef maybe_log_error
 }
 
+uint64_t sandstone_user_cpu_time(uint64_t from)
+{
+#ifdef __unix__
+    struct rusage time;
+    if (getrusage(RUSAGE_THREAD, &time) < 0) {
+        return 0;
+    }
+    return (1000000LL * time.ru_utime.tv_sec) + time.ru_utime.tv_usec - from;
+#else
+    return 0;
+#endif
+}
+
+uint64_t sandstone_sys_cpu_time(uint64_t from)
+{
+#ifdef __unix__
+    struct rusage time;
+    if (getrusage(RUSAGE_THREAD, &time) < 0) {
+        return 0;
+    }
+    return (1000000LL * time.ru_stime.tv_sec) + time.ru_stime.tv_usec - from;
+#else
+    return 0;
+#endif
+}
+
+uint64_t sandstone_wall_clock_time(uint64_t from)
```

Comment:
I think this should be moved to `sandstone_chrono.h`.

Reply:
'chrono' sounds good.